### PR TITLE
Add support for uploading compiled firmware to the qmk bucket

### PR DESCRIPTION
**Pull request by skullydazed**
This uploads compiled firmware to https://qmk.nyc3.digitaloceanspaces.com/ which will eventually let us remove firmware files from qmk.fm.

### CODE REVIEW

File: qmk_storage.py [hunk 1]
```diff
@@ -41,14 +42,16 @@
...
-except Exception as e:
-    if e.__class__.__name__ not in ['BucketAlreadyOwnedByYou', 'BucketAlreadyExists']:
-        logging.warning('Could not contact S3! Storage related functionality will not work!')
+for bucket in [S3_BUCKET, COMPILE_S3_BUCKET]:
+    try:
+        s3.create_bucket(Bucket=bucket)
+
+    except Exception as e:
+        if e.__class__.__name__ not in ['BucketAlreadyOwnedByYou', 'BucketAlreadyExists']:
+            logging.warning('Could not contact S3! Storage related functionality will not work!')
```
Comment (Erovia) [inline qmk_storage.py:51]:
```suggestion
    
    except (botocore.exceptions.BucketAlreadyOwnedByYou, botocore.exceptions.BucketAlreadyExists) as e:
        logging.debug('Bucket already exists.')
    except Exception as e:
        logging.warning('Could not contact S3! Storage related functionality will not work!')
```

**Review by Erovia — CHANGES_REQUESTED** [review]

File: qmk_storage.py (hunk 1 above)
Comment (Erovia) [reply qmk_storage.py:51]: I think [line 8](https://github.com/qmk/qmk_compiler/blob/public_firmware/qmk_storage.py#L8) could also be just `import botocore.exceptions`.

Comment (skullydazed) [reply qmk_storage.py:51]: Did you actually test this? When I first wrote the code I couldn't find anything that would actually match the exception as thrown.

Comment (Erovia) [reply qmk_storage.py:51]: You are right, these exceptions are not directly exposed on the client side.
After digging through the docs, I found [the recommended way to catch these](https://botocore.amazonaws.com/v1/documentation/api/latest/client_upgrades.html#error-handling), so updated the comment.

File: qmk_storage.py [hunk 2]
```diff
@@ -41,14 +42,19 @@
...
-    if e.__class__.__name__ not in ['BucketAlreadyOwnedByYou', 'BucketAlreadyExists']:
-        logging.warning('Could not contact S3! Storage related functionality will not work!')
+for bucket in [S3_BUCKET, COMPILE_S3_BUCKET]:
+    try:
+        s3.create_bucket(Bucket=bucket)
+
+    
+    except ClientError as e:
+        if e.response['Error']['Code'] in ['BucketAlreadyOwnedByYou', 'BucketAlreadyExists'] and DEBUG > 1:
+            logging.debug('Bucket already exists.')
```
Comment (skullydazed) [inline qmk_storage.py:52]:
```suggestion
    except ClientError as e:
        if e.response['Error']['Code'] in ['BucketAlreadyOwnedByYou', 'BucketAlreadyExists'] and DEBUG > 1:
            logging.debug('Bucket %s already exists.', bucket)
```

**Review by Erovia — APPROVED** [review]

**Pull request merged**